src/StackExchange.Redis/RedisKey.cs

```diff
@@ -297,5 +298,59 @@ internal static RedisKey WithPrefix(byte[]? prefix, RedisKey value)
         /// </summary>
         /// <param name="suffix">The suffix to append.</param>
         public RedisKey Append(RedisKey suffix) => WithPrefix(this, suffix);
+
+
+        internal bool TryGetSimpleBuffer([NotNullWhen(true)] out byte[]? arr)
+        {
+            arr = KeyValue is null ? Array.Empty<byte>() : KeyValue as byte[];
+            return arr is not null && (KeyPrefix is null || KeyPrefix.Length == 0);
+        }
+
+        internal int TotalLength()
+            => (KeyPrefix is null ? 0 : KeyPrefix.Length) + KeyValue switch
+                {
+                    null => 0,
+                    string s => Encoding.UTF8.GetByteCount(s),
+                    _ => ((byte[])KeyValue).Length,
+                };
+
+        internal int CopyTo(Span<byte> destination)
+        {
+            int written = 0;
+            if (KeyPrefix is not null && KeyPrefix.Length != 0)
+            {
+                KeyPrefix.CopyTo(destination);
+                written += KeyPrefix.Length;
+                destination = destination.Slice(KeyPrefix.Length);
+            }
+            switch (KeyValue)
+            {
+                case null:
+                    break; // nothing to do
+                case string s:
+                    if (s.Length != 0)
+                    {
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_1_OR_GREATER
```

Comment:
Since we only target `netstandard2.0`, simplify to `#if NETCOREAPP`?

Reply:
Noted; I assume `NETCOREAPP` includes 5.0+?

Reply:
Aye, it's _not_ netstandard but all .NET Core onwards including 5/6/7, etc.